cassovary-core/src/main/scala/com/twitter/cassovary/util/collections/CSeq.scala

```diff
@@ -0,0 +1,158 @@
+package com.twitter.cassovary.util.collections
+
+import scala.collection.immutable
+
+trait CSeq[@specialized(Int, Long) T] {
```

Comment:
pls add some comments for the reason of existence (brief summary of what we discussed I guess)